server/src/main/java/org/cloud/network/router/deployment/VpcRouterDeploymentDefinition.java

```diff
@@ -115,12 +111,55 @@ protected void checkPreconditions() {
      */
     @Override
     protected boolean prepareDeployment() {
-        return true;
+        isPublicNetwork = needsPublicNic();
+
+        boolean canProceed = true;
```

Comment:
This bit comes from here: https://github.com/MissionCriticalCloud/cosmic-core/blob/3eeb5f20c1b686c6a35b9e8dfa660bf4d9cbc3d5/server/src/main/java/org/cloud/network/router/deployment/RouterDeploymentDefinition.java#L330-L336